### PR TITLE
50-kafka-recordbatch-must-contain-at-least-one-record

### LINKED ISSUE #50: Kafka RecordBatch must contain at least one record
According to the [documentation](https://kafka.apache.org/documentation/#messageformat), the Kafka RecordBatch must contain at least one record (either control or standard record):

> The technical term for a batch of messages is a record batch, **and a record batch contains one or more records.**

 This behavior is also reflected in the [Java Kafka Clients](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java#L582). 
 
Please fix the `RecordBatch` class to not allow for creating batches with less than 1 record.

### PR DESCRIPTION
**Pull request by Kalin8900**
closes #50

### CODE REVIEW

File: src/protocol/primitives/record-batch/compressed-records.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { type Compressor } from '../../compression';
+import { InvalidRecordBatchError, NullInNonNullableFieldError } from '../../exceptions';
+import { ReadBuffer, WriteBuffer } from '../../serialization';
+import { Array } from '../array';
+import { Int32 } from '../int32';
+import { Record } from './record';
+
+export class CompressedRecords {
```
Comment (kordiangrabowski) [inline src/protocol/primitives/record-batch/compressed-records.ts:8]: Apparently the `CompressedArray` was too generic. I like this change.

File: src/protocol/primitives/record-batch/record-batch.spec.ts [hunk 2]
```diff
@@ -239,9 +247,55 @@ describe('RecordBatch', () => {
     const deserialized = await RecordBatch.deserialize(new ReadBuffer(buffer));
 
     expect(deserialized).toBeDefined();
-    expect(deserialized.records.value).toHaveLength(1);
-    expect(deserialized.records.value![0]!.key.value).toEqual(Buffer.from([0x01, 0x02, 0x03, 0x04, 0x05, 0x06]));
-    expect(deserialized.records.value![0]!.value.value).toEqual(Buffer.from([0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c]));
+    expect(deserialized.records).toHaveLength(1);
+    expect(deserialized.records![0]!.key.value).toEqual(Buffer.from([0x01, 0x02, 0x03, 0x04, 0x05, 0x06]));
```
Comment (kordiangrabowski) [inline src/protocol/primitives/record-batch/record-batch.spec.ts:251]: is the `deserialized.records!` non-null assertion operator needed here?

File: src/protocol/primitives/record-batch/record-batch.spec.ts [hunk 3]
```diff
@@ -239,9 +247,55 @@ describe('RecordBatch', () => {
...
+        producerEpoch: new Int16(Int16.MIN_VALUE),
+        baseSequence: new Int32(Int32.MIN_VALUE),
+        records: []
+      })
+    ).toThrow();
+  });
+
+  it('should throw if deserialized length of records is smaller than 0', async () => {
+    const buffer = Buffer.from(
+      '80000000000000000000003a8000000002ce38005a005480000000800000000000000080000000000000008000000000000000800080000000ffffffff28b52ffd2000010000',
```
Comment (kordiangrabowski) [inline src/protocol/primitives/record-batch/record-batch.spec.ts:281]: Is there a reason for introducing the new style for providing the buffers?

Comment (Kalin8900) [reply src/protocol/primitives/record-batch/record-batch.spec.ts:281]: Laziness

**Comment by Kalin8900** [comment]
@dependabot rebase

**Review by kordiangrabowski — DISMISSED** [review]

**Review by kordiangrabowski — APPROVED** [review]

**Pull request merged**